airflow/utils/email.py

```diff
@@ -241,27 +241,33 @@ def send_mime_email(
     smtp_user = None
     smtp_password = None
 
+    found_email_connection = False
     if conn_id is not None:
         try:
             from airflow.hooks.base import BaseHook
 
             airflow_conn = BaseHook.get_connection(conn_id)
             smtp_user = airflow_conn.login
             smtp_password = airflow_conn.password
+            found_email_connection = True
         except AirflowException:
             pass
-    if smtp_user is None or smtp_password is None:
-        warnings.warn(
-            "Fetching SMTP credentials from configuration variables will be deprecated in a future "
-            "release. Please set credentials using a connection instead.",
-            RemovedInAirflow3Warning,
-            stacklevel=2,
-        )
+    if not found_email_connection:
         try:
             smtp_user = conf.get("smtp", "SMTP_USER")
             smtp_password = conf.get("smtp", "SMTP_PASSWORD")
         except AirflowConfigException:
-            log.debug("No user/password found for SMTP, so logging in with no authentication.")
+            pass
+        # if at this point smtp_user or smtp_password is set it came from conf.get, trigger warning
+        if smtp_user is not None or smtp_password is not None:
+            warnings.warn(
+                "Fetching SMTP credentials from configuration variables will be deprecated in a future "
+                "release. Please set credentials using a connection instead.",
+                RemovedInAirflow3Warning,
+                stacklevel=2,
+            )
+    if smtp_user is None or smtp_password is None:
+        log.debug("No user/password found for SMTP, so logging in with no authentication.")
```

Comment:
Why move this one?

Reply:
Because its a valid warning that no creds were supplied in the connection and will be using non authentication smtp

Reply:
No I mean does it not do its job at its original location? Why not?

Reply:
We want it to trigger after ether a connection has been found or failback to airflow.cfg variables have been parsed

So i moved it after the above had been completed